modules/claims_api/lib/custom_error.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class CustomError
+    def initialize(error, claim, method)
+      @error = error
+      @claim = claim
+      @method = method
+    end
+
+    def build_error # rubocop:disable Metrics/MethodLength
+      if @error == Faraday::ConnectionFailed || @error == Faraday::ParsingError
+        custom_error = { 'messages' => [{ 'key' => 'ServiceException',
+                                          'detail' => 'A Faraday error has occurred, original_error: ' \
+                                                      "#{@error}.", status: '500' }] }
+        update_claim(custom_error)
+        raise EVSS::DisabilityCompensationForm::ServiceException, custom_error
+
+      elsif @error.is_a?(::Common::Exceptions::BackendServiceException)
+        custom_error = { 'messages' => [{ 'key' => 'BackendException',
+                                          'detail' => 'A backend exception occurred, original_error: ' \
+                                                      "#{@error}.", status: '500' }] }
+        update_claim(custom_error)
+        raise EVSS::DisabilityCompensationForm::ServiceException, custom_error
+      elsif @error.is_a?(StandardError)
+        custom_error = { 'messages' => [{ 'key' => 'Client error',
+                                          'detail' => 'A client exception has occurred, original_error: ' \
+                                                      "#{@error}.", status: '400' }] }
+        update_claim(custom_error)
+        raise ::Common::Exceptions::BadRequest
+      else
+        custom_error = { 'messages' => [{ 'key' => 'Unknown error',
+                                          'detail' => 'An unknown error has occurred, original_error: ' \
+                                                      "#{@error}.", status: '500' }] }
+        update_claim(custom_error)
+        raise @error
+      end
+    end
+
+    def update_claim(custom_error)
+      log_outcome_for_claims_api(custom_error)
+      @claim.status = ClaimsApi::AutoEstablishedClaim::ERRORED
+      @claim.evss_response = custom_error['messages']
```

Comment:
@stiehlrod  I believe the evss_response  is getting saved over when we raise the error and go back into the sidekiq job. Stepping through and it does go here as expected but the evss_response that gets saved here (below)

<img width="1060" alt="Screenshot 2024-02-28 at 10 07 18 AM" src="https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/94754658-1d1b-4d24-a12b-e71bb58da53c">

If I try to GET the claim with this evss_response it gets returned
![Screenshot 2024-02-28 at 10 10 39 AM](https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/8744ff82-3afc-44d3-a08c-ca628d0a755a)

However the process doe snot end there and it appears the evss_response get saved over when it raises and goes back to the sidekiq job.  At the end of the whole thing I see an evss_response of 

<img width="547" alt="Screenshot 2024-02-28 at 10 09 43 AM" src="https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/e66f79e1-d8e9-41b1-b187-2008e95413e8">

This does not return the claim, but shows the claim not found error after it saves here
![Screenshot 2024-02-28 at 10 10 53 AM](https://github.com/department-of-veterans-affairs/vets-api/assets/129893414/f52a3622-e886-4662-b432-842e4885bfe8)